arangod/Replication2/ReplicatedLog/Components/CompactionManager.cpp

```diff
@@ -116,7 +116,18 @@ void CompactionManager::triggerAsyncCompaction(
     guard->_compactionInProgress = true;
     while (true) {
       ADB_PROD_ASSERT(guard.isLocked());
-      auto store = guard->storage.transaction();
+
+      auto storeRes = basics::catchToResultT([&]() -> auto {
+        return guard->storage.transaction();
+      });
+      if (storeRes.fail()) {
+        // The participant might be gone
+        LOG_CTX("d7724", DEBUG, self->loggerContext)
+            << "error during compaction: " << storeRes.result();
+        break;
+      }
```

Comment:
You should report the error, roughly as follows:
```suggestion
      if (storeRes.fail()) {
        // The participant might be gone
        LOG_CTX("d7724", ERR, self->loggerContext)
            << "error during compaction: " << storeRes.result();
        cresult.error = std::move(storeRes).result().error();
        guard->status.lastCompaction->error = cresult.error;
        clearCompactionRunning.fire();
        break;
      }
```
Please double-check the code, it was written blindly.

Reply:
Addressed in https://github.com/arangodb/arangodb/pull/20628/commits/444c825fbff19e5f971ef36f315a0c7c2003a184